### PR TITLE
Add new FCU API support in Outscale EC2 driver

### PR DESCRIPTION
**Pull request by jmunhoz**
Add libcloud support for the following new APIs available in the Flexible Computing Unit (FCU)
- DescribeInstanceTypes
- DescribeProductTypes
- GetProductType
- DescribeQuota
- ModifyInstanceKeypair

The patchset includes tests and fixtures for the new APIs.

Signed-off-by: Javier M. Mellid jmunhoz@igalia.com

### CODE REVIEW

File: libcloud/compute/drivers/ec2.py [hunk 1]
```diff
@@ -6806,6 +6811,268 @@ def list_sizes(self, location=None):
...
+        :param      marker: Set quota marker
+        :type       marker: ``string``
+
+        :return:    (is_truncated, quota) tuple
+        :rtype:     ``(bool, dict)``
+        """
+
+        if filters:
+            raise NotImplementedError(
+                'quota filters are not implemented')
```
Comment (tonybaloney) [inline libcloud/compute/drivers/ec2.py:6912]: when will this be implemented?

**Review by tonybaloney — COMMENTED** [review]
Great thanks @jmunhoz for the updates.

File: libcloud/compute/drivers/ec2.py (hunk 1 above)
Comment (jmunhoz) [reply libcloud/compute/drivers/ec2.py:6912]: @tonybaloney thanks for reviewing. we hope a new release of FCU at the end of November or December. Filters in this API are being reworked cause the original implementation is raising some non expected issues in the server side and we had to rollback although the API doc was published and the interface spec is stable. This way users get a clear message and they would not report this well-know bug.

BTW this first patchset would be also a first update to catch up with the new features and improvements in the FCU. Outscale would like to contribute new APIs and available features implemented along the previous months in the server side. As expected, with each new FCU release we would revisit the non implemented bits and bugfixes to support the new features as soon as they hit production in the server side.

**Comment by tonybaloney** [comment]
Ok. I'm happy 👍

**Comment by jmunhoz** [comment]
@tonybaloney thanks! :+1:

**Pull request closed without merging**